### DISCUSSION TITLE
Field localization

### DISCUSSION THREAD

**Discussion by Adelysnet** [sulu/sulu#6636] [Q&A] [answered]
Hi,
I see that I may be possible de make a field non translatable : 

`Reuse a field in different localizations by defining it as non-translatable`
 https://sulu.io/product/solutions/multilingual-websites

How to achieve that ? is there any tag to add to the template ?

Thanks !

**Comment by alexander-schranz** [comment, marked as answer]
For root properties in your template you can set multingual false. See the template documentation: https://docs.sulu.io/en/2.4/book/templates.html#language-independent-properties
The default value is true so translatable. The property does not work inside blocks. So only for root properties.

**Comment by Adelysnet** [comment]
Great !

Missed that..

Thanks !

**Comment by Adelysnet** [comment]
Is there any way to migrate existing data ? 
Adding the multilangual:false property to my block removes the content in my default locale.

**Reply by alexander-schranz** [reply]
You need to create a phpcr migration. See following like as example: https://docs.sulu.io/en/2.4/cookbook/migrate-content-data.html

**Comment by Adelysnet** [comment]
I had to make some minors changes to make migrations working : 

Add session and localization argument to the getRowsToMigrate method : 
```
- private function getRowsToMigrate()
+ private function getRowsToMigrate(SessionInterface $session, Localization $localization)
```

Moved localization iteration to the downgrade/upgrade methods : 
```
private function getRowsToMigrate(SessionInterface $session, Localization $localization)
    {
        $queryManager = $session->getWorkspace()->getQueryManager();
        - $localizations = $this->container->get('sulu_core.webspace.webspace_manager')->getAllLocalizations();
```
```
 /**
     * Upgrade all nodes in given session.
     */
    private function upgrade(SessionInterface $session)
    {
        + $localizations = $this->container->get('sulu_core.webspace.webspace_manager')->getAllLocalizations();

        + /** @var Localization $localization */
        + foreach ($localizations as $localization) {

            /** @var Row $row */
            foreach ($this->getRowsToMigrate($session, $localization) as $row) { ...
```

**Reply by alexander-schranz** [reply]
Seems like they did got lost here: https://github.com/sulu/sulu-docs/pull/706 can you create a pull request for the sulu docs 2.3 branch? /cc @mamazu

**Reply by mamazu** [reply]
Ups, yeah my mistake. Didn't check it properly. That of course needs to exist.

**Comment by Adelysnet** [comment]
Is there any way to migrate existing data ? Adding the multilangual:false property to my block removes the content in my default locale.

Romain Ezquerra
Développement Web

Tél. :05.57.95.82.87



Bureaux : 11 cours de la somme -33000 Bordeaux
www.adelysnet.com
